src/formatter.rs

```diff
@@ -83,13 +117,119 @@ where
                 command = title;
                 debug!("Detected command name: {}", &command);
             }
-            LineType::Description(text) => println!("  {}", config.style.description.paint(text)),
-            LineType::ExampleText(text) => println!("  {}", config.style.example_text.paint(text)),
+            LineType::Description(text) => yield_snippet(HighlightingSnippet::Description(&text)),
+            LineType::ExampleText(text) => yield_snippet(HighlightingSnippet::Text(&text)),
             LineType::ExampleCode(text) => {
-                println!("      {}", &format_code(&command, &text, &config))
+                yield_snippet(HighlightingSnippet::NormalCode("      "));
+                highlight_code(&command, &text, &mut yield_snippet);
+                yield_snippet(HighlightingSnippet::Linebreak);
             }
+
             LineType::Other(text) => debug!("Unknown line type: {:?}", text),
         }
     }
-    println!();
+    yield_snippet(HighlightingSnippet::Linebreak);
+}
+
+fn print_snippet<'a>(snip: HighlightingSnippet<'a>, style: &StyleConfig) {
+    use HighlightingSnippet::*;
+
+    match snip {
+        CommandName(s) => print!("{}", style.command_name.paint(s)),
+        Variable(s) => print!("{}", style.example_variable.paint(s)),
+        NormalCode(s) => print!("{}", style.example_code.paint(s)),
+        Description(s) => println!("  {}", style.description.paint(s)),
+        Text(s) => println!("  {}", style.example_text.paint(s)),
+        Linebreak => println!(),
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use HighlightingSnippet::*;
+
+    #[test]
+    fn test_is_freestanding_substring() {
+        let t = |s: &str, words: &[(usize, usize)], expected: bool| {
```

Comment:
This local function is only helpful for line 160 in that it iterates over the array of sub-strings. In test code it is sometimes better to be a bit more verbose instead of hiding functionality somewhere. As a reader I would have liked

    t("I love tldr", (0, 1), true);
    t("I love tldr", (2, 6), true);
    t("I love tldr", (7, 11), true);

more which would have avoided the `&[...]`s in addition.

Reply:
Good point :+1:

Reply:
Actually, I could remove `t` altogether :thinking: